src/server/api/controllers/vessel.controller.js

```diff
@@ -0,0 +1,118 @@
+"use strict";
+
+const HttpError = require("../lib/utils/http-error");
+const knex = require("../../config/db");
+
+//GET
+const getVessels = async req => {
+  try {
+    return await knex("vessels").select(
+      "vessels.id",
+      "vessels.organization_id",
+      "vessels.mmsi",
+      "vessels.name"
+    );
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+//GET:id
+const getVesselById = async id => {
+  try {
+    const vessels = await knex("vessels")
+      .select("vessels.id as id", "organization_id", "mmsi", "name")
+      .where({ id: id });
+    if (vessels.length === 0) {
+      throw new HttpError(
+        "Bad request",
+        `Cannot find vessel for ID ${id}!`,
+        404
+      );
+    }
+    return vessels;
+  } catch (err) {
+    return err.message;
+  }
+};
+//POST
+
+const createVessel = async ({ body }) => {
+  const { organization_id, mmsi, name } = body;
+
+  const organization = await knex
+    .from("organizations")
+    .select("*")
+    .where({
+      id: organization_id
+    });
+
+  if (organization.length === 0) {
+    throw new HttpError(
+      "Bad request",
+      `${organization_id} organization not found!`,
+      404
+    );
+  }
+
+  const vessels = await knex("vessels").insert({
+    organization_id: organization[0].id,
```

Comment:
It seems strange and convoluted to use `organization[0].id` when you also has a variable called `organization_id` that contains the same value.

Why would you use the ID of the record you just fetched from the DB by matching it with the ID in your request if you could use the ID from your request?

If you use the ID from the request, the object you are inserting with knex looks like this:
```
{
  organization_id: organization_id,
  mmsi: mmsi,
  name, name
}
```
But since both the keys and the names of the variables you use as values are the same, why not use destructuring?
```
{ organization_id, mmsi, name }
```

Same output, more concise and easier to read :)

Reply:
Agree, and avoid accessing array object properties using `[0].id` it can lead to headaches later

Reply:
Used restructuring to read easy